cloudinit/config/cc_ca_certs.py

```diff
@@ -44,60 +44,110 @@
 from cloudinit import subp
 from cloudinit import util
 
-CA_CERT_PATH = "/usr/share/ca-certificates/"
-CA_CERT_FILENAME = "cloud-init-ca-certs.crt"
-CA_CERT_CONFIG = "/etc/ca-certificates.conf"
-CA_CERT_SYSTEM_PATH = "/etc/ssl/certs/"
-CA_CERT_FULL_PATH = os.path.join(CA_CERT_PATH, CA_CERT_FILENAME)
-
-distros = ['alpine', 'debian', 'ubuntu']
+DISTRO_DEFAULT_CONFIG = {
+    'ubuntu': {
+        'ca_cert_path': '/usr/share/ca-certificates/',
+        'ca_cert_filename': 'cloud-init-ca-certs.crt',
+        'ca_cert_config': '/etc/ca-certificates.conf',
+        'ca_cert_system_path': '/etc/ssl/certs/',
+        'ca_cert_update_exe': 'update-ca-certificates'
+    },
+    'debian': {
+        'ca_cert_path': '/usr/share/ca-certificates/',
+        'ca_cert_filename': 'cloud-init-ca-certs.crt',
+        'ca_cert_config': '/etc/ca-certificates.conf',
+        'ca_cert_system_path': '/etc/ssl/certs/',
+        'ca_cert_update_exe': 'update-ca-certificates'
+    },
+    'alpine': {
+        'ca_cert_path': '/usr/share/ca-certificates/',
+        'ca_cert_filename': 'cloud-init-ca-certs.crt',
+        'ca_cert_config': '/etc/ca-certificates.conf',
+        'ca_cert_system_path': '/etc/ssl/certs/',
+        'ca_cert_update_exe': 'update-ca-certificates'
+    },
+    'rhel': {
+        'ca_cert_path': '/usr/share/pki/ca-trust-source/',
+        'ca_cert_filename': 'anchors/cloud-init-ca-certs.crt',
+        'ca_cert_config': '',
+        'ca_cert_system_path': '/etc/pki/ca-trust/',
+        'ca_cert_update_exe': 'update-ca-trust'
+    }
+}
+
+
+distros = ['alpine', 'debian', 'ubuntu', 'rhel']
+
+
+def _distro_ca_certs_configs(distro):
+    """Construct a distro-specific ca_certs config dictionary by merging
+       distro specific changes into base config.
+
+    @param distro: String providing the distro class name.
+    @returns: Dict of distro configurations for ntp clients.
+    """
+    dcfg = DISTRO_DEFAULT_CONFIG
+    if distro in dcfg:
+        cfg = DISTRO_DEFAULT_CONFIG[distro]
+        cfg['ca_cert_full_path'] = os.path.join(cfg['ca_cert_path'],
+                                                cfg['ca_cert_filename'])
+    else:
+        cfg = {}
+    return cfg
 
 
-def update_ca_certs():
+def update_ca_certs(distro_name):
     """
     Updates the CA certificate cache on the current machine.
     """
-    subp.subp(["update-ca-certificates"], capture=False)
+    distro_cfg = _distro_ca_certs_configs(distro_name)
+    subp.subp([distro_cfg['ca_cert_update_exe']], capture=False)
 
 
-def add_ca_certs(certs):
+def add_ca_certs(distro_name, certs):
     """
     Adds certificates to the system. To actually apply the new certificates
     you must also call L{update_ca_certs}.
 
     @param certs: A list of certificate strings.
     """
+    distro_cfg = _distro_ca_certs_configs(distro_name)
     if certs:
```

Comment:
You didn't add this line, but as you're already modifying this entire function, inverting this to:

```py
if not certs:
    return
```

would allow you to dedent everything else in here by one step.

Reply:
(that's one of my favourite things to do)